### PR TITLE
Adding custom mouse cursors for flame

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Adds a small new feature to `GameWidget` and `Game`: The ability to have and change during runtime the cursor of the mouse.

More features can be built on top of this in the future, like a `Draggable` equivalent for mouse cursors on components.

https://user-images.githubusercontent.com/835641/132077015-1f7c23d3-e80b-4347-a44d-1018b8298cbd.mov

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: packages/flame/lib/src/game/game.dart [hunk 1]
```diff
@@ -250,3 +255,20 @@ class ActiveOverlaysNotifier extends ChangeNotifier {
   /// Returns if the given [overlayName] is active
   bool isActive(String overlayName) => _activeOverlays.contains(overlayName);
 }
+
+/// A [ChangeNotifier] used to control the mouse cursor of the GameWidget which is running this game
+class GameMouseCursorNotifier extends ChangeNotifier {
```
Comment (st-pasha) [inline packages/flame/lib/src/game/game.dart:260]: Isn't this exactly the same as `ValueNotifier<MouseCursor>`?

File: doc/gesture-input.md [hunk 2]
```diff
@@ -102,6 +102,22 @@ and [MouseRegion widget](https://api.flutter.dev/flutter/widgets/MouseRegion-cla
 also read more about Flutter's gestures
 [here](https://api.flutter.dev/flutter/gestures/gestures-library.html).
 
+It is also possible to change the current mouse cursor displayed on the `GameWidget` region. To so
```
Comment (spydon) [inline doc/gesture-input.md:105]: To do so*

File: doc/gesture-input.md [hunk 3]
```diff
@@ -102,6 +102,22 @@ and [MouseRegion widget](https://api.flutter.dev/flutter/widgets/MouseRegion-cla
...
+just the following code can be used inside the `Game` class
```
Comment (spydon) [inline doc/gesture-input.md:106]: -just

**Review by wolfenrain — APPROVED** [review]

File: packages/flame/lib/src/game/game.dart (hunk 1 above)
Comment (erickzanardo) [reply packages/flame/lib/src/game/game.dart:260]: You are correct, totally forgot about this helper class.

File: doc/gesture-input.md (hunk 2 above)
Comment (luanpotter) [reply doc/gesture-input.md:105]: `To do so,`**

File: doc/gesture-input.md (hunk 3 above)
Comment (luanpotter) [reply doc/gesture-input.md:106]:
```suggestion
the following code can be used inside the `Game` class.
```

File: packages/flame/lib/src/game/game.dart [hunk 4]
```diff
@@ -209,6 +209,11 @@ abstract class Game extends Projector {
   /// - GameWidget
   /// - [Game.overlays]
   final overlays = ActiveOverlaysNotifier();
+
+  /// Used to change the mouse cursor of the GameWidget running this game
+  /// when setting the value to null will make the GameWidget defer the choice
+  /// of the cursor to the closest region available on the tree
```
Comment (luanpotter) [inline packages/flame/lib/src/game/game.dart:216]:
```suggestion
  /// Used to change the mouse cursor of the GameWidget running this game.
  /// Setting the value to null will make the GameWidget defer the choice
  /// of the cursor to the closest region available on the tree.
```

**Review by luanpotter — APPROVED** [review]
Nice!

**Pull request merged**